portfolio/src/main/java/com/google/sps/servlets/DeleteData.java

```diff
@@ -0,0 +1,40 @@
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.StreamSupport;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+/** Servlet that deletes all of the comment data. */
+@WebServlet("/delete-data")
+public class DeleteData extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Iterable<Entity> resultsIterable = results.asIterable();
+
+    StreamSupport.stream(resultsIterable.spliterator(), false)
+        .map(entity->entity.getKey())
+        .forEach(datastore::delete);
+  }
```

Comment:
nit: I would suggest deleter(Transaction, Iterable<>) instead of delete(Key key)
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete-com.google.appengine.api.datastore.Transaction-java.lang.Iterable-

This is good practice to make sure that delete works for all the keys.

Reply:
I tried using transaction but I kept getting `java.lang.IllegalArgumentException: operating on too many entity groups in a single transaction.` even though the only entity group I'm using is Comment. 

Since this is a nit, would it be okay if I leave it as is using just stream?

Reply:
For context, the code I wrote to attempt this is: 
```
    TransactionOptions options = TransactionOptions.Builder.withXG(true);
    Transaction txn = datastore.beginTransaction(options);

    Iterable<Key> keys = StreamSupport.stream(resultsIterable.spliterator(), false)
        .map(entity->entity.getKey())
        .collect(Collectors.toList());

    datastore.delete(txn, keys);
    txn.commit();
```